README.md

```diff
@@ -1,53 +1,103 @@
+### Example 1: Basic formatting with all non-zero units
 
-### Example 1: Greet an entity
+This example shows the default behavior of `Format-TimeSpan`, which converts TimeSpan objects into human-readable formatted strings showing all units that have non-zero values.
 
-Provide examples for typical commands that a user would like to do with the module.
+```powershell
+New-TimeSpan -Hours 2 -Minutes 30 -Seconds 10 | Format-TimeSpan -Format FullName
+# Output: 2 hours 30 minutes 10 seconds
+```
+
+### Example 2: Including zero values
+
+Use the `IncludeZeroValues` parameter to include units with zero values in the output, showing all time units from the highest non-zero unit down to the smallest unit.
 
 ```powershell
-Greet-Entity -Name 'World'
-Hello, World!
+New-TimeSpan -Hours 2 -Minutes 30 -Seconds 10 | Format-TimeSpan -Format FullName -IncludeZeroValues
+# Output: 2 hours 30 minutes 10 seconds 0 milliseconds 0 microseconds
 ```
 
-### Example 2
+### Example 3: Using different format styles
+
+```powershell
+# Symbol format (default)
+New-TimeSpan -Minutes 90 | Format-TimeSpan
+# Output: 1h 30m
 
-Provide examples for typical commands that a user would like to do with the module.
+# Abbreviation format
+New-TimeSpan -Minutes 90 | Format-TimeSpan -Format Abbreviation
+# Output: 1hr 30min
+
+# Full name format
+New-TimeSpan -Minutes 90 | Format-TimeSpan -Format FullName
+# Output: 1 hour 30 minutes
+```
+
+### Example 4: Explicit precision (backward compatibility)
 
 ```powershell
-Import-Module -Name PSModuleTemplate
+# Show only the most significant unit (rounded)
+New-TimeSpan -Minutes 90 | Format-TimeSpan -Precision 1
+# Output: 2h
+
+# Show specific number of units
+[TimeSpan]::FromSeconds(3661) | Format-TimeSpan -Precision 2 -Format FullName
+# Output: 1 hour 1 minute
+```
+
+### Example 5: Force specific base unit
+
+```powershell
+[TimeSpan]::FromMinutes(2) | Format-TimeSpan -BaseUnit 'Seconds'
+# Output: 120s
 ```
 
 ### Find more examples
 
 To find more examples of how to use the module, please refer to the [examples](examples) folder.
 
-Alternatively, you can use the Get-Command -Module 'This module' to find more commands that are available in the module.
-To find examples of each of the commands you can use Get-Help -Examples 'CommandName'.
+Alternatively, you can use `Get-Command -Module TimeSpan` to find more commands that are available in the module.
+To find examples of each of the commands you can use `Get-Help -Examples Format-TimeSpan`.
+
+## Parameters
+
+### Format-TimeSpan
+
+- **TimeSpan** (Required): The TimeSpan object to format
+- **Precision** (Optional): Number of units to include in output. If not specified, shows all non-zero units
+- **BaseUnit** (Optional): Force formatting from a specific unit (e.g., 'Seconds', 'Minutes')
+- **Format** (Optional): Display format - 'Symbol' (default), 'Abbreviation', or 'FullName'
+- **IncludeZeroValues** (Switch): Include units with zero values in the output
+
```

Comment:
Lets remove this, as this is function specific and covered in the function documentation.